app/src/main/java/com/example/nockanakalinowej/Model/GameController.java

```diff
@@ -50,6 +68,12 @@ public void setLevel(int levelNo) {
         currentLevel = levels[levelNo];
     }
 
+    public void restartLevel(){
+        currentLevel.setTilesIDs(availableTilesIDs);
+        currentLevel.shuffleTiles();
+        currentLevel.setVariant(2 - currentLevel.getVariant() + 1);

```

Comment:
change to:
currentLevel.setVariant(currentLevel.getVariant()%GameProperties.variantsNo +1);